### PR TITLE
fix: remove incremental done steps

### PR DESCRIPTION
**Pull request by carenthomas**
**Please describe the purpose of this pull request.**
don't stream done tokens in between multi step requests, only support the final done token

**How to test**
How can we test your PR during review? What commands should we run? What outcomes should we expect?
poetry run pytest -s -vv tests/test_client_legacy.py::test_streaming_send_message

**Have you tested this PR?**
Have you tested the latest commit on the PR? If so please provide outputs from your tests.

**Related issues or PRs**
Please link any related GitHub [issues](https://github.com/letta-ai/letta/issues) or [PRs](https://github.com/letta-ai/letta/pulls).

**Is your PR over 500 lines of code?**
If so, please break up your PR into multiple smaller PRs so that we can review them quickly, or provide justification for its length.

**Additional context**
Add any other context or screenshots about the PR here.

### CODE REVIEW

File: letta/server/rest_api/routers/v1/agents.py [hunk 1]
```diff
@@ -729,14 +729,14 @@ async def send_message_to_agent(
...
+                    isinstance(message, LettaMessage) or isinstance(message, LegacyLettaMessage) or message == OPENAI_SSE_DONE
                 ), type(message)
                 generated_stream.append(message)
-                if message == MessageStreamStatus.done:
+                if message == OPENAI_SSE_DONE:
                     break
 
             # Get rid of the stream status messages
-            filtered_stream = [d for d in generated_stream if not isinstance(d, MessageStreamStatus)]
+            filtered_stream = [d for d in generated_stream if d != OPENAI_SSE_DONE]
```
Comment (carenthomas) [inline letta/server/rest_api/routers/v1/agents.py:739]: what is this filter step for?

File: tests/test_client_legacy.py [hunk 2]
```diff
@@ -245,45 +246,35 @@ def test_streaming_send_message(mock_e2b_api_key_none, client: RESTClient, agent
...
     send_message_ran = False
-    # 3. Check that we get all the start/stop/end tokens we want
-    #    This includes all of the MessageStreamStatus enums
-    done_gen = False
-    done_step = False
+    # 3. Check that we get the end token we want (StreamDoneStatus)
     done = False
 
-    # print(response)
+    print(response)
```
Comment (carenthomas) [inline tests/test_client_legacy.py:252]: will remove

File: letta/client/streaming.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 from letta.constants import OPENAI_CONTEXT_WINDOW_ERROR_SUBSTRING
 from letta.errors import LLMError
-from letta.schemas.enums import MessageStreamStatus
+from letta.llm_api.openai import OPENAI_SSE_DONE
```
Comment (carenthomas) [inline letta/client/streaming.py:9]: I'm using the constant for the equality checks and my new literal type for the type hint on the union used for the rv

File: letta/server/rest_api/interface.py [hunk 4]
```diff
@@ -381,9 +377,6 @@ def stream_end(self):
         """Clean up the stream by deactivating and clearing chunks."""
         self.streaming_chat_completion_mode_function_name = None
 
-        if not self.streaming_chat_completion_mode and not self.nonstreaming_legacy_mode:
-            self._push_to_buffer(self.multi_step_gen_indicator)
```
Comment (carenthomas) [inline letta/server/rest_api/interface.py:385]: making the assumption that we want to never stream the intermediate done tokens

**Review by carenthomas — COMMENTED** [review]
a little complicated because this value is used in the LettaStreamingResponse return value which is a union of different types, so wasn't able to just directly use the constant everywhere. this has the implication that LettaStreamingResponse can no longer be used in isinstance checks since StreamDoneStatus is not a class

**Comment by cpacker** [comment]
Example `main`:

```
curl -X POST \
    -H 'Content-Type: application/json' \
    -H 'Accept: text/event-stream' \
    -d '{
  "messages": [
    {
      "role": "user",
      "text": "hello there!"
    }
  ],
  "stream_steps": true,
  "stream_tokens": true
}' \
    http://localhost:8283/v1/agents/agent-6272c3d2-c33b-4ead-8cc7-372df841ea09/messages/stream
```
```
data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":"User"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" has"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" just"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" logged"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" in"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" and"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" greeted"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" me"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":"."}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" Let's"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" keep"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" this"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" friendly"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" and"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":" engaging"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"reasoning_message","reasoning":"!"}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"name":"send_message","tool_call_id":"2ea15c97-2824-46a2-9c91-eae62"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":"{\"message\":\""}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":"Hey"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" there"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":"!"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" It's"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" great"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" to"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" have"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" you"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" here"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":"."}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" How"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" can"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" I"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" help"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" you"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":" today"}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":"?\""}}

data: {"id":"message-9c703ecd-6ca9-4069-a3fe-690a38f9ec5e","date":"2024-12-21T00:47:23+00:00","message_type":"tool_call_message","tool_call":{"arguments":"}"}}

data: [DONE_GEN]

data: {"id":"message-0213b36e-5b1c-4f8c-be60-77cff90a2ebb","date":"2024-12-21T00:47:25+00:00","message_type":"tool_return_message","tool_return":"None","status":"success","tool_call_id":"2ea15c97-2824-46a2-9c91-eae62","stdout":null,"stderr":null}

data: [DONE_STEP]

data: {"usage":{"completion_tokens":45,"prompt_tokens":2620,"total_tokens":2665,"step_count":1}}

data: [DONE]
```

**Comment by cpacker** [comment]
Example `caren/let-591-move-to-using-only-done-for-the-streaming-api`

```
data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":"User"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" repeated"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" their"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" greeting"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":"."}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" I"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" should"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" acknowledge"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" it"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" and"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" encourage"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" a"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":" conversation"}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"reasoning_message","reasoning":"."}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"name":"send_message","tool_call_id":"14b6e622-a80d-4dc7-a8f6-c282d"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"{\"message\":\"Hello"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" again"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"!"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" What's"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" on"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" your"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" mind"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"?"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" I'm"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" here"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" to"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" help"}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":".\""}}

data: {"id":"message-dbdeac83-808a-4bb0-bb9b-0cbb70c11ec2","date":"2024-12-21T00:48:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"}"}}

data: {"id":"message-fb544286-9cee-47a1-b7d7-702400be487a","date":"2024-12-21T00:48:58+00:00","message_type":"tool_return_message","tool_return":"None","status":"success","tool_call_id":"14b6e622-a80d-4dc7-a8f6-c282d","stdout":null,"stderr":null}

data: {"id":"null","date":"2024-12-21T00:48:58+00:00","message_type":"usage_message","usage":{"completion_tokens":38,"prompt_tokens":2821,"total_tokens":2859,"step_count":1}}

data: [DONE]
```

**Comment by cpacker** [comment]
```
data: {"id":"message-fb544286-9cee-47a1-b7d7-702400be487a","date":"2024-12-21T00:48:58+00:00","message_type":"tool_return_message","tool_return":"None","status":"success","tool_call_id":"14b6e622-a80d-4dc7-a8f6-c282d","stdout":null,"stderr":null}

data: {"id":"null","date":"2024-12-21T00:48:58+00:00","message_type":"usage_message","usage":{"completion_tokens":38,"prompt_tokens":2821,"total_tokens":2859,"step_count":1}}

data: [DONE]
```

Only comment: both `id` and `date` should be nullable, and we should omit `null`s on the SSE emit.

**Comment by cpacker** [comment]
Patched:
```
data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":"User"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" keeps"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" greeting"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" me"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":"."}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" They"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" might"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" be"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" testing"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" responses"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" or"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" looking"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" for"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":" engagement"}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"reasoning_message","reasoning":"."}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"name":"send_message","tool_call_id":"bd1d6c9d-e10e-4dc0-bac9-3a2cf"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"{\"message\":\"Hey"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" there"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" again"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"!"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" I'm"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" really"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" glad"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" to"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" see"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" you"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"!"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" How"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" can"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" I"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" assist"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" you"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":" today"}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"?\""}}

data: {"id":"message-adace973-b750-421c-8bff-b43eff19d27b","date":"2024-12-21T00:58:57+00:00","message_type":"tool_call_message","tool_call":{"arguments":"}"}}

data: {"id":"message-4e700b4a-d3a6-42ff-b10e-6bd95d8898a0","date":"2024-12-21T00:58:58+00:00","message_type":"tool_return_message","tool_return":"None","status":"success","tool_call_id":"bd1d6c9d-e10e-4dc0-bac9-3a2cf","stdout":null,"stderr":null}

data: {"message_type":"usage_message","usage":{"completion_tokens":44,"prompt_tokens":3016,"total_tokens":3060,"step_count":1}}

data: [DONE]
```

File: tests/test_client_legacy.py [hunk 5]
```diff
@@ -245,45 +247,35 @@ def test_streaming_send_message(mock_e2b_api_key_none, client: RESTClient, agent
...
+                # Some rough metrics for a reasonable usage pattern
+                assert chunk.usage.step_count == 1
+                assert chunk.usage.completion_tokens > 10
+                assert chunk.usage.prompt_tokens > 1000
+                assert chunk.usage.total_tokens > 1000
+        elif chunk == OPENAI_SSE_DONE:
+            assert not done, "Message stream already done"
+            done = True
+        else:
+            assert isinstance(chunk, LettaStreamingResponse)
```
Comment (cpacker) [inline tests/test_client_legacy.py:272]: @carenthomas why did you move the `assert` into an `else`? Can we keep it before the `if isinstance(chunk, LettaMessage):`?

**Comment by cpacker** [comment]
Deprecated in favor of #2297

**Comment by carenthomas** [comment]
Re-implemented here: https://github.com/letta-ai/letta-cloud/pull/504
